File: app/core/users_results/default_recoding_values/multiple_choice.rb
# frozen_string_literal: true

module UsersResults
  module DefaultRecodingValues
    class MultipleChoice < Base
      def perform
        (0...question.props['choices']).map { |i| { 'index' => i, 'value' => i + 1 } }
      end
    end
  end
end

File: app/frontend/modules/reports/models/QuestionsTranslates/MultipleChoice.js
import _ from 'lodash'
import BaseTranslate from './BaseTranslate'

class MultipleChoice extends BaseTranslate {
  getValueByCode (field, extraData) {
    if (field === 'questionText') {
      return this.question.props.questionText
    }
    if (/^choicesTexts/.test(field)) {
      return this.question.props.choicesTexts[extraData.choice]
    }
  }

  exportLocales () {
    const result = {
      questionText: this.question.props.questionText,
    }
    _.times(this.question.props.choices, (i) => {
      result[`choicesTexts${i + 1}`] = this.question.props.choicesTexts[i]
    })
    return result
  }
}

export default MultipleChoice
